hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```diff
@@ -27,19 +25,18 @@
 public abstract class AbstractGrid extends Grid {
 
     private static final long serialVersionUID = 4856562746502217630L;
-    
-    @Autowired
-    protected I18N i18n;
 
-    @Autowired
-    protected transient EventBus.SessionEventBus eventBus;
+    protected final I18N i18n;
+
+    protected final transient EventBus.UIEventBus eventBus;
 
+    protected final SpPermissionChecker permissionChecker;
+
+    protected AbstractGrid(final I18N i18n, final UIEventBus eventBus, final SpPermissionChecker permissionChecker) {
```

Comment:
I think these kind of things are hard to standardise. I would guess that none of our constructors follow any pattern here.